RestartManager.js

```diff
@@ -0,0 +1,33 @@
+let NativeCodePush = require("react-native").NativeModules.CodePush;
+
+const RestartManager = (() => {
+    let _allowed = true;
+
+    function restartApp(onlyIfUpdateIsPending = false) {
+        if (_allowed) {
+            NativeCodePush.restartApp(onlyIfUpdateIsPending);
+        }
+    }
+
+    function allow() {
```

Comment:
@scottbommarito I agree we would need to accommodate that scenario, but that doesn't mean we shouldn't account for resume-based installs as well. If we're going to have a feature for disallowing synthetic restarts, it doesn't make sense for it to be specific to updates installed via `IMMEDIATE`, since that is just one way of triggering it.

The `RestartManager` could easily have a private method called `clearPendingRestart` (or whatever) that is called anytime the `LocalPackage.install` method is called with a non-immediate mode.